packages/perseus-editor/src/components/__tests__/locked-function-settings.test.tsx

```diff
@@ -0,0 +1,365 @@
+            await userEvent.click(colorSwitch);
+            const colorOption = screen.getByText("green");
+            await userEvent.click(colorOption);
+
+            // Assert
+            expect(onChangeProps).toHaveBeenCalledWith({color: "green"});
+        });
+
+        test("calls 'onChangeProps' when stroke style is changed", async () => {
+            // Arrange
+            render(
+                <LockedFunctionSettings
+                    {...defaultProps}
+                    expanded={true}
+                    onChangeProps={onChangeProps}
+                />,
+                {wrapper: RenderStateRoot},
+            );
+
+            // Act
+            // Change the stroke
+            const strokeSwitch = screen.getByLabelText("stroke");
+            await userEvent.click(strokeSwitch);
+            const strokeOption = screen.getByText("dashed");
+            await userEvent.click(strokeOption);
+
+            // Assert
+            expect(onChangeProps).toHaveBeenCalledWith({strokeStyle: "dashed"});
+        });
+
+        test("calls 'onChangeProps' when equation is changed (as keys are pressed)", async () => {
+            // Arrange
+            render(
+                <LockedFunctionSettings
+                    {...defaultProps}
+                    expanded={true}
+                    onChangeProps={onChangeProps}
+                    equation=""
+                />,
+                {wrapper: RenderStateRoot},
+            );
+
+            // Act
+            // Change the equation
+            const equationField = screen.getByLabelText("equation");
+            await userEvent.type(equationField, "zot");
+
+            // Assert
+            expect(onChangeProps).toHaveBeenCalledTimes(3);
+            expect(onChangeProps).toHaveBeenNthCalledWith(1, {equation: "z"});
+            expect(onChangeProps).toHaveBeenNthCalledWith(2, {equation: "o"});
+            expect(onChangeProps).toHaveBeenNthCalledWith(3, {equation: "t"});
+        });
+
+        test("calls 'onChangeProps' when directional axis is changed", async () => {
+            // Arrange
+            const onChangeProps = jest.fn();
+            render(
+                <LockedFunctionSettings
+                    {...defaultProps}
+                    expanded={true}
+                    onChangeProps={onChangeProps}
+                />,
+                {wrapper: RenderStateRoot},
+            );
+
+            // Act
+            // Change the axis direction
+            const directionalAxisSwitch =
+                screen.getByLabelText("equation prefix");
+            await userEvent.click(directionalAxisSwitch);
+            const axisOption = screen.getByText("x =");
+            await userEvent.click(axisOption);
+
+            // Assert
+            expect(onChangeProps).toHaveBeenCalledWith({directionalAxis: "y"});
+        });
+
+        describe("Domain restrictions", () => {
```

Comment:
Is this describe block supposed to be nested within the "settings interactions" describe block?

Reply:
Yes. Since domain restrictions have more tests, I wanted them grouped. Since these tests are still interaction tests, I kept them as a sub-block within "Settings interactions". Based upon your question, I've changed the title to be "Domain interactions" to be more consistent.